Lib/fontTools/varLib/instancer.py

```diff
@@ -196,59 +200,108 @@ def _getVarRegionScalar(location, regionAxes):
 
 
 def _scaleVarDataDeltas(varData, regionScalars):
-    # multiply all varData deltas in-place by the corresponding region scalar
-    varRegionCount = len(varData.VarRegionIndex)
-    scalars = [regionScalars[regionIndex] for regionIndex in varData.VarRegionIndex]
+    # multiply all varData deltas in-place by the corresponding region scalar, and
+    # remove regions whose scalar is 0
+    scalars = [regionScalars[ri] for ri in varData.VarRegionIndex]
     for item in varData.Item:
-        assert len(item) == varRegionCount
-        item[:] = [delta * scalar for delta, scalar in zip(item, scalars)]
+        item[:] = [
+            delta * scalar for delta, scalar in zip(item, scalars) if scalar != 0
+        ]
+    varData.VarRegionIndex = [
+        ri for ri in varData.VarRegionIndex if regionScalars[ri] != 0
+    ]
+    varData.VarRegionCount = len(varData.VarRegionIndex)
 
 
-def _getVarDataDeltasForRegions(varData, regionIndices, rounded=False):
-    # Get only the deltas that correspond to the given regions (optionally, rounded).
-    # Returns: list of lists of float
+def _popVarDataDeltas(varData, regionIndex):
+    # For each item's delta-set row, pop the column that correspond to regionIndex.
+    # If the region is not referenced by the varData.VarRegionIndex, return all zeros
+    # and keep varData as is.
+    # Returns: list of deltas, one per VarData item.
```

Comment:
general comment: some of these comments are explaining *what* we are doing (which one can generally tell from the code itself) and not *why* (which is harder to know from the code)